### PR TITLE
feat: add share button for events

### LINKED ISSUE #1721: Add links that let users join chapters and attend events
Currently, if I follow a link like http://localhost:3000/events/22?emaillink=true I will see a modal asking me to confirm that would like to join the event.

If would be helpful for administrators to be able to easily share this after creating a new event.  For example, the dashboard for that event could have a share button that shows the link and gives the user an option to copy it to their clipboard.

Similarly, if they want to invite people to join their Chapter it would be helpful to be able to do the same sort of thing from the Chapter's dashboard.

Nice to have:

Logged out users should be sent away to log in and then brought back to the shared link if they consent.

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Ref #1721

### CODE REVIEW

File: client/src/modules/dashboard/Events/components/Actions.tsx [hunk 1]
```diff
@@ -72,6 +73,9 @@ const Actions: React.FC<ActionsProps> = ({
       <Button size="sm" colorScheme="blue" onClick={clickEmailAttendees}>
         Email Attendees
       </Button>
+      <ShareEventPop
+        link={`${process.env.CLIENT_LOCATION}/events/${event.id}?emaillink=true`}
```
Comment (ojeytonwilliams) [inline client/src/modules/dashboard/Events/components/Actions.tsx:77]:
```suggestion
        link={`${process.env.NEXT_PUBLIC_CLIENT_URL}/events/${event.id}?emaillink=true`}
```
Sorry, the env vars are needlessly complicated. I shouldn't have stored `CLIENT_LOCATION` in two variables.

Anyway, this is the one you need.

File: client/src/modules/dashboard/Events/components/ShareEventPopOver.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          Share
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent>
+        <PopoverArrow />
+        <PopoverCloseButton />
+        <PopoverHeader>
+          You have copied the link below into your clipboard!
+        </PopoverHeader>
+        <PopoverBody>{link}</PopoverBody>
```
Comment (ojeytonwilliams) [inline client/src/modules/dashboard/Events/components/ShareEventPopOver.tsx:31]: How about not showing the link?  It's a little prettier without it!

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Comment by ojeytonwilliams** [comment]
@Sboonny were you planning to handle the 'share chapter' link in this PR?

**Comment by Sboonny** [comment]
I want the discussion about the join and subscribing to chapter to end first, because I don't know what people want to share. 

Are they going to ask people to follow (subscirbe) to their chapter, or to join it?

But I may be lazy and add both of them, but if users grow attach to a feature removing it would be annoying :think:

To be fair, I may be overthinking this, this feels like I am overthinking it.

**Comment by ojeytonwilliams** [comment]
> Are they going to ask people to follow (subscirbe) to their chapter, or to join it?

I'd just mirror what we do for events.  So, a link that triggers the "are you sure" modal for joining a chapter (though we should update the text, since we're triggering it automatically).

As for subscriptions, that's still being discussed https://github.com/freeCodeCamp/chapter/issues/1639 but this PR doesn't have to worry since it's just triggering the current behaviour.  i.e. making a request to the `joinChapter` resolver and it handles subscriptions.

**Comment by ojeytonwilliams** [comment]
> I am not sure if the link works.

It won't, yet, but we can handle that in another PR.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: Thanks, @Sboonny!

**Pull request merged**